code/game/gamemodes/trifecta/trifecta.dm

```diff
@@ -101,12 +103,58 @@
 /datum/game_mode/trifecta/post_setup()
 	for(var/datum/mind/vampire as anything in pre_vampires)
 		vampire.add_antag_datum(/datum/antagonist/vampire)
+
 	for(var/datum/mind/changeling as anything in pre_changelings)
 		changeling.add_antag_datum(/datum/antagonist/changeling)
+
 	for(var/datum/mind/traitor as anything in pre_traitors)
-		traitor.add_antag_datum(/datum/antagonist/traitor)
+		var/datum/antagonist/traitor/tot_datum = new()
+		tot_datum.delayed_objectives = TRUE
+		traitor.add_antag_datum(tot_datum)
+
+	if(length(pre_traitors))
+		var/random_time = rand(5 MINUTES, 15 MINUTES)
+		addtimer(CALLBACK(src, PROC_REF(late_handout)), random_time)
+
 	..()
 
+/datum/game_mode/trifecta/proc/traitors_to_add()
+	var/extra_points = num_players_started() - cost_at_roundstart
+	if(extra_points - TOT_COST < 0)
+		return 0 // Not enough new players to add extra tots
+
+	. = 0
+	while(extra_points)
+		.++
+		if(extra_points < TOT_COST) // The leftover change is enough for us to buy another traitor with, what a deal!
+			return
+		extra_points -= TOT_COST
+
+/datum/game_mode/trifecta/late_handout()
```

Comment:
I'm not really happy with this being duped over the two gamemodes. `reveal_traitor_objectives` is already scoped to `game_mode` so this one could possibly also be scoped there. Not super clean though since it is traitor specific code.